File: src/ffi/ffi_bridge.c
/**
 * ffi_bridge.c — C++ / Rust FFI köprüsü
 * Ring buffer fonksiyonları Rust tarafından implement edilir.
 * Bu dosya C++ tarafının kullandığı stub tanımlarını içerir.
 *
 * Not: ffi_auto.h C++ syntax'ı içerdiğinden buraya include edilmez.
 * RJ_FFI_VERSION burada yerel olarak tanımlanır; ffi_bridge.h ile aynı değer olmalı.
 */

#include <stdint.h>

#define RJ_FFI_VERSION_LOCAL 0x00010000u  /* 1.0.0 — ffi_bridge.h ile eşleşmeli */

/* Versiyon kontrolü */
uint32_t rj_ffi_version(void) {
    return RJ_FFI_VERSION_LOCAL;
}

File: src/ffi/ffi_bridge.zig
const std = @import("std");

// ABI doğrulama — derleme zamanında
comptime {
    // RjMetricSample boyutu 56 byte olmalı
    // ffi_auto.h'dan import edilecek
}

export fn rj_ffi_version() u32 {
    return 0x0001_0000; // v1.0.0 — ffi_bridge.h RJ_FFI_VERSION ile eşleşmeli
}

// V8/I14: rj_metrics_poll stub'ı kaldırıldı. Gerçek implementasyon artık
// Rust orchestrator'da (src/orchestrator/src/ffi.rs) — MetricState'ten pull.
// Burada bir stub bırakmak, aynı sembolü iki statik kütüphanede tanımlayıp
// linkte duplicate-symbol (LNK2005) üretirdi.
